devlab/tests/functional_test.py

```diff
@@ -31,3 +41,52 @@ class FunctionalTest(unittest.TestCase):
     def __init__(self, *args, **kwargs):
         super(FunctionalTest, self).__init__(*args, **kwargs)
         suppress_dependency_logging()
+        self.src_cloud = Prerequisites(cloud_prefix='SRC')
+        self.dst_cloud = Prerequisites(cloud_prefix='DST')
+        self.filtering_utils = FilteringUtils()
+
+    def filter_resources(self, obj):
```

Comment:
**Much** better would be to have a separate `filter` method for each type of object, any method **must not** deal with objects of different type. This solution not only complicates implementation, it's also completely unreadable, hardly maintainable and difficult to read.